analyzer/src/main/kotlin/managers/Gradle.kt

```diff
@@ -77,9 +69,33 @@ class Gradle(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfig
                 Gradle(analyzerConfig, repoConfig)
     }
 
-    val maven = MavenSupport { localRepositoryManager ->
-        GradleLocalRepositoryManager(localRepositoryManager)
-    }
+    val maven = MavenSupport(object : WorkspaceReader {
+        private val workspaceRepository = WorkspaceRepository()
+        private val gradleCacheRoot = File(System.getProperty("user.home"), ".gradle/caches/modules-2/files-2.1")
+
+        override fun findArtifact(artifact: Artifact): File? {
+            val artifactRootDir = File(gradleCacheRoot,
+                    "${artifact.groupId}/${artifact.artifactId}/${artifact.version}")
+
+            val artifactFile = artifactRootDir.walkTopDown().find {
+                val classifier = if (artifact.classifier.isNullOrBlank()) "" else "${artifact.classifier}-"
+                it.isFile && it.name == "${artifact.artifactId}-$classifier${artifact.version}.${artifact.extension}"
+            }
+
+            log.debug {
+                "Gradle cache result for '${artifact.identifier()}:${artifact.classifier}:${artifact.extension}': " +
+                        artifactFile?.absolutePath
+            }
+
+            return artifactFile
```

Comment:
Could use the pattern we also use in some other places when adding logging for the return value:

```
return artifactRootDir.walkTopDown()
    .find { ... }
    .also { log.debug { ... } }
```

Reply:
In this case I find the current code more readable, mainly because the log output is quite log.